Blog: KEP-3314 CSI CBT beta announcement blog (#54615)
kubernetes/website: content/en/blog/_posts/2026-xx-xx-csi-cbt-moves-beta.md

---
layout: blog

> title: 'Moving Changed Block Tracking API support to Beta'

Reviewer: Because this article is now very old, try this:
```suggestion
title: 'Kubernetes Changed Block Tracking API - Beta Differences`
```

draft: true
slug: csi-changed-block-tracking-beta
author: >
   [Prasad Ghangal](https://github.com/PrasadG193) (Veeam Kasten)
---

Changed Block Tracking (CBT) support for CSI drivers
[shipped as Alpha](/blog/2025/09/25/csi-changed-block-tracking/) in
September 2025. With the `v1.0.0` release of the
[external-snapshot-metadata](https://github.com/kubernetes-csi/external-snapshot-metadata)

> project, the feature now moves to **Beta**.

Reviewer: ```suggestion
Changed Block Tracking (CBT) support for CSI drivers
[shipped as Alpha](/blog/2025/09/25/csi-changed-block-tracking/) in
September 2025. With the March `v1.0.0` release of the
[external-snapshot-metadata](https://github.com/kubernetes-csi/external-snapshot-metadata)
project, the feature moved to **Beta**.
```

> If you are new to changed block tracking, the

Reviewer: ```suggestion
If you aren't yet familiar with _changed block tracking_ for storage in
Kubernetes, the
```

[Alpha announcement](/blog/2025/09/25/csi-changed-block-tracking/) covers
the motivation, the three primary components (the CSI `SnapshotMetadata`
gRPC service, the SnapshotMetadataService CRD, and the
`external-snapshot-metadata` sidecar), and a walkthrough of how to use the
API. This post focuses on what is different in Beta.

## What's new in Beta

The main change in this release is the promotion of the
SnapshotMetadataService CRD from `v1alpha1` to `v1beta1`. The CRD used to
advertise a driver's metadata service now serves
`cbt.storage.k8s.io/v1beta1`. The schema itself is unchanged, but this

> release removes `v1alpha1` rather than serving it alongside the new version.

Reviewer: ```suggestion
release **removes** `v1alpha1` (rather than serving it alongside the new version).
```
